packages/nx/src/command-line/nx-commands.ts

```diff
@@ -735,7 +735,7 @@ function withGenerateOptions(yargs: yargs.Argv) {
       default: false,
     })
     .middleware((args) => {
-      if (process.env.NX_INTERACTIVE !== 'false') {
+      if (process.env.NX_INTERACTIVE === 'true') {
         args.interactive = true;
       } else {
```

Comment:
`--interactive` is on by default right? So this means

`NX_INTERACTIVE=false nx g app` would still prompt?

I think something like this would be more accurate:

```suggestion
      if (process.env.NX_INTERACTIVE === 'false') {
        args.interactive = false;
      } else {
```

With this we get the following behavior:
- NX_INTERACTIVE=true => args.interactive is true due to default.
- NX_INTERACTIVE=false => args.interactive is false due to override.
- --interactive=true or --interactive=false => NX_INTERACTIVE set appropriately.